### DISCUSSION TITLE
Menu disappears when changing keyboard layout while searching

### DISCUSSION THREAD

**Discussion by Skeevert** [lxqt/lxqt#2285] [Ideas]
Greetings. I have noticed that lxqt-menu disappears when I try to change layout while searching, and it's kinda annoying. I have managed to track it down to  lxqt-panel/plugin-mainmenu/lxqtmainmenu.cpp
LXQtMainMenu::eventFilter actually has a code which closes menu if any of the key combinations are pressed.
I've come up with an idea to query setxkbmap to determine key combination to change layout (just like in a  lxqt-config/lxqt-config-input/keyboardlayoutconfig.cpp) and then somehow compare it to the keys pressed.
Is there any better way of doing it? And is it even worth it?

**Comment by Skeevert** [comment]
Or maybe add an option to ignore key combinations (except for ESC) while searching?

**Reply by tsujan** [reply]
If you mean https://github.com/lxqt/lxqt-panel/issues/1717, it's already fixed (in https://github.com/lxqt/lxqt-panel/commit/229516303b1c69a053af538a21191280b30ff64f) and released.

**Reply by Skeevert** [reply]
Oh, I see. It turns out I have an old lxqt. Thanks!
